File: frontend/src/utils/iconMapping.ts
import { colors } from '@/config/colors'

// Icon mapping for food items
export const getFoodIcon = (type: string, category: string): string => {
  // Map by specific food types first
  const typeIcons: Record<string, string> = {
    // Chicken
    'chicken': 'mdi:food-drumstick',
    'wings': 'mdi:food-drumstick',
    'breast': 'mdi:food-drumstick',
    
    // Beef
    'beef': 'mdi:food-steak',
    'burger': 'fluent-emoji:hamburger',
    'steak': 'mdi:food-steak',
    
    // Pizza
    'pizza': 'twemoji:pizza',
    
    // Fish
    'fish': 'mdi:fish',
    'salmon': 'mdi:fish',
    'cod': 'mdi:fish',
    
    // Sides
    'fries': 'noto:french-fries',
    'potato': 'mdi:food-french-fries',
    'rice': 'mdi:rice',
    'salad': 'noto:green-salad',
    'vegetables': 'mdi:food-apple',
    
    // Drinks
    'soda': 'mdi:bottle-soda-classic',
    'cola': 'mdi:bottle-soda-classic',
    'water': 'mdi:cup-water',
    'beer': 'mdi:beer',
    'wine': 'mdi:glass-wine',
    'coffee': 'mdi:coffee',
    'tea': 'mdi:tea',
    
    // Desserts
    'ice-cream': 'mdi:ice-cream',
    'cake': 'mdi:cake-variant',
    'cookie': 'mdi:cookie',
    
    // Sauces
    'sauce': 'game-icons:ketchup',
    'ketchup': 'game-icons:ketchup',
    'mayo': 'game-icons:ketchup',
    'mustard': 'game-icons:ketchup',
    
    // Additional types
    'snack': 'mdi:food-variant',
    'fruit': 'mdi:food-apple',
    'sub': 'game-icons:sandwich',
    'sandwich': 'mdi:food-hot-dog',
    'wrap': 'mdi:food-hot-dog',
    'soup': 'mdi:food-variant',
    'curry': 'mdi:food-variant',
    'kebab': 'mdi:food-variant',
    'hotdog': 'mdi:food-hot-dog',
    'panini': 'mdi:food-hot-dog',
    'quesadilla': 'mdi:food-variant',
    'burrito': 'mdi:food-variant',
    'enchilada': 'mdi:food-variant',
    'fajita': 'mdi:food-variant',
    'gyro': 'mdi:food-variant',
    'shawarma': 'mdi:food-variant',
    'falafel': 'mdi:food-variant',
    'dumpling': 'mdi:food-variant',
    'springroll': 'mdi:food-variant',
    'eggroll': 'mdi:food-variant',
    'tempura': 'mdi:food-variant',
    'teriyaki': 'mdi:food-variant',
    'stirfry': 'mdi:food-variant',
    'lo mein': 'mdi:food-variant',
    'pad thai': 'mdi:food-variant',
    'pho': 'mdi:food-variant',
    'ramen': 'mdi:food-variant',
    'udon': 'mdi:food-variant',
    'soba': 'mdi:food-variant',
    'bibimbap': 'mdi:food-variant',
    'bulgogi': 'mdi:food-variant',
    'japchae': 'mdi:food-variant',
    'kimchi': 'mdi:food-variant',
    'taco': 'mdi:food-variant',
    'thai': 'mdi:food-variant',
    'noodlesoup': 'mdi:food-variant',
    'noodles': 'mdi:food-variant',
    'pasta': 'mdi:food-variant',
    'sushi': 'mdi:food-variant',
    
    // Generic by category
    'main': 'mdi:food',
    'side': 'mdi:food-fork-drink',
  }

  // Try to find by type first
  const typeKey = type.toLowerCase()
  if (typeIcons[typeKey]) {
    return typeIcons[typeKey]
  }

  // Fallback to category
  const categoryKey = category.toLowerCase()
  if (typeIcons[categoryKey]) {
    return typeIcons[categoryKey]
  }

  // Default fallback
  return 'mdi:food'
}

// Get icon color based on food type and category
export const getFoodIconColor = (type: string, category: string): string => {
  // Map by specific food types first
  const typeColors: Record<string, string> = {
    // Chicken - golden brown
    'chicken': '#D4AF37',
    'wings': '#D4AF37',
    'breast': '#D4AF37',
    
    // Beef - red/brown
    'beef': '#8B4513',
    'burger': '#8B4513',
    'steak': '#8B4513',
    
    // Pizza - orange/red
    'pizza': '#FF6B35',
    
    // Fish - blue
    'fish': '#4A90E2',
    'salmon': '#FF6B6B',
    'cod': '#4A90E2',
    
    // Sides - green/brown
    'fries': '#8B4513',
    'potato': '#8B4513',
    'rice': '#F4A460',
    'salad': '#228B22',
    'vegetables': '#228B22',
    
    // Drinks - blue/purple
    'soda': '#4A90E2',
    'cola': '#8B0000',
    'water': '#87CEEB',
    'beer': '#FFD700',
    'wine': '#8B0000',
    'coffee': '#8B4513',
    'tea': '#228B22',
    
    // Desserts - pink/purple
    'ice-cream': '#FF69B4',
    'cake': '#FF69B4',
    'cookie': '#D2691E',
    'dessert': '#FF69B4',
    
    // Sauces - orange/red
    'sauce': '#FF4500',
    'ketchup': '#FF0000',
    'mayo': '#FFFFF0',
    'mustard': '#FFD700',
  }

  // Try to find by type first
  const typeKey = type.toLowerCase()
  if (typeColors[typeKey]) {
    return typeColors[typeKey]
  }

  // Fallback to category colors
  const categoryColors: Record<string, string> = {
    'main': colors.accent,
    'side': colors.success,
    'drink': colors.info,
    'dessert': colors.pink,
  }

  const categoryKey = category.toLowerCase()
  if (categoryColors[categoryKey]) {
    return categoryColors[categoryKey]
  }

  // Default fallback
  return colors.primary
}

// Get icon size class based on context
export const getIconSizeClass = (size: 'sm' | 'md' | 'lg' = 'md'): string => {
  switch (size) {
    case 'sm':
      return 'w-4 h-4'
    case 'md':
      return 'w-5 h-5'
    case 'lg':
      return 'w-6 h-6'
    default:
      return 'w-5 h-5'
  }
} 

// Get main food type from offer
export const getMainFoodType = (offer: { main_items?: Array<{ type: string; quantity?: number }> }): { type: string; displayName: string; icon: string } => {
  // Check if we have main items
  if (offer.main_items && offer.main_items.length > 0) {
          // If multiple main items, create a combined display
      if (offer.main_items.length > 1) {
        const uniqueTypes = [...new Set(offer.main_items.map(item => item.type.toLowerCase()))]
        
                // Check if all items are the same type (e.g., multiple burgers)
        const allSameType = uniqueTypes.length === 1
        const totalQuantity = offer.main_items.reduce((sum, item) => sum + (item.quantity || 1), 0)
        
        // Map food types to display names
        const typeDisplayNames: Record<string, string> = {
          'chicken': 'Chicken',
          'wings': 'Chicken Wings',
          'breast': 'Chicken Breast',
          'beef': 'Beef',
          'burger': 'Burger',
          'steak': 'Steak',
          'pizza': 'Pizza',
          'fish': 'Fish',
          'salmon': 'Salmon',
          'cod': 'Cod',
          'taco': 'Taco',
          'thai': 'Thai',
          'noodlesoup': 'Noodle Soup',
          'noodles': 'Noodles',
          'pasta': 'Pasta',
          'sushi': 'Sushi',
          'sandwich': 'Sandwich',
          'wrap': 'Wrap',
          'salad': 'Salad',
          'soup': 'Soup',
          'curry': 'Curry',
          'kebab': 'Kebab',
          'hotdog': 'Hot Dog',
          'sub': 'Sub',
          'panini': 'Panini',
          'quesadilla': 'Quesadilla',
          'burrito': 'Burrito',
          'enchilada': 'Enchilada',
          'fajita': 'Fajita',
          'gyro': 'Gyro',
          'shawarma': 'Shawarma',
          'falafel': 'Falafel',
          'dumpling': 'Dumpling',
          'springroll': 'Spring Roll',
          'eggroll': 'Egg Roll',
          'tempura': 'Tempura',
          'teriyaki': 'Teriyaki',
          'stirfry': 'Stir Fry',
          'friedrice': 'Fried Rice',
          'lo mein': 'Lo Mein',
          'pad thai': 'Pad Thai',
          'pho': 'Pho',
          'ramen': 'Ramen',
          'udon': 'Udon',
          'soba': 'Soba',
          'bibimbap': 'Bibimbap',
          'bulgogi': 'Bulgogi',
          'japchae': 'Japchae',
          'kimchi': 'Kimchi',
          'snack': 'Snack',
          'fruit': 'Fruit',
          'cookie': 'Cookie',
          'dessert': 'Dessert',
          'ice-cream': 'Ice Cream',
          'cake': 'Cake',
          'drink': 'Drink',
          'soda': 'Soda',
          'water': 'Water',
          'coffee': 'Coffee',
          'tea': 'Tea',
          'beer': 'Beer',
          'wine': 'Wine',
          'sauce': 'Sauce',
          'ketchup': 'Ketchup',
          'mayo': 'Mayo',
          'mustard': 'Mustard',
          'fries': 'Fries',
          'potato': 'Potato',
          'rice': 'Rice',
          'vegetables': 'Vegetables'
        }
        
        // Create combined display name
        if (allSameType && totalQuantity > 1) {
          // If all items are the same type and total quantity > 1, show quantity
          const primaryType = uniqueTypes[0]
          const displayName = `${totalQuantity} ${typeDisplayNames[primaryType] || primaryType.charAt(0).toUpperCase() + primaryType.slice(1)}s`
          const icon = getFoodIcon(primaryType, 'main')
          return { type: primaryType, displayName, icon }
        } else {
          // Different types or single item
          const displayNames = uniqueTypes.map(type => 
            typeDisplayNames[type] || type.charAt(0).toUpperCase() + type.slice(1)
          )
          const displayName = displayNames.join(' + ')
          
          // Use the first type for icon (or a generic combo icon)
          const primaryType = uniqueTypes[0]
          const icon = getFoodIcon(primaryType, 'main')
          
          return { type: primaryType, displayName, icon }
        }
    } else {
      // Single main item - use original logic
      const mainItem = offer.main_items[0]
      const type = mainItem.type.toLowerCase()
      
      // Map food types to display names
      const typeDisplayNames: Record<string, string> = {
        'chicken': 'Chicken',
        'wings': 'Chicken Wings',
        'breast': 'Chicken Breast',
        'beef': 'Beef',
        'burger': 'Burger',
        'steak': 'Steak',
        'pizza': 'Pizza',
        'fish': 'Fish',
        'salmon': 'Salmon',
        'cod': 'Cod',
        'taco': 'Taco',
        'thai': 'Thai',
        'noodlesoup': 'Noodle Soup',
        'noodles': 'Noodles',
        'pasta': 'Pasta',
        'sushi': 'Sushi',
        'sandwich': 'Sandwich',
        'wrap': 'Wrap',
        'salad': 'Salad',
        'soup': 'Soup',
        'curry': 'Curry',
        'kebab': 'Kebab',
        'hotdog': 'Hot Dog',
        'sub': 'Sub',
        'panini': 'Panini',
        'quesadilla': 'Quesadilla',
        'burrito': 'Burrito',
        'enchilada': 'Enchilada',
        'fajita': 'Fajita',
        'gyro': 'Gyro',
        'shawarma': 'Shawarma',
        'falafel': 'Falafel',
        'dumpling': 'Dumpling',
        'springroll': 'Spring Roll',
        'eggroll': 'Egg Roll',
        'tempura': 'Tempura',
        'teriyaki': 'Teriyaki',
        'stirfry': 'Stir Fry',
        'friedrice': 'Fried Rice',
        'lo mein': 'Lo Mein',
        'pad thai': 'Pad Thai',
        'pho': 'Pho',
        'ramen': 'Ramen',
        'udon': 'Udon',
        'soba': 'Soba',
        'bibimbap': 'Bibimbap',
        'bulgogi': 'Bulgogi',
        'japchae': 'Japchae',
        'kimchi': 'Kimchi',
        'snack': 'Snack',
        'fruit': 'Fruit',
        'cookie': 'Cookie',
        'dessert': 'Dessert',
        'ice-cream': 'Ice Cream',
        'cake': 'Cake',
        'drink': 'Drink',
        'soda': 'Soda',
        'water': 'Water',
        'coffee': 'Coffee',
        'tea': 'Tea',
        'beer': 'Beer',
        'wine': 'Wine',
        'sauce': 'Sauce',
        'ketchup': 'Ketchup',
        'mayo': 'Mayo',
        'mustard': 'Mustard',
        'fries': 'Fries',
        'potato': 'Potato',
        'rice': 'Rice',
        'vegetables': 'Vegetables'
      }
      
      const displayName = typeDisplayNames[type] || type.charAt(0).toUpperCase() + type.slice(1)
      const icon = getFoodIcon(type, 'main')
      
      return { type, displayName, icon }
    }
  }
  
  // Fallback if no main items
  return { type: 'food', displayName: 'Mat', icon: 'mdi:food' }
}

File: scraper/src/icon_mapping.py
"""
Icon Mapping System for Fastfood Scraper
Handles all icon assignments and data processing logic
"""

from typing import Dict, List, Any, Optional
import re
import json
from pathlib import Path


class IconMapping:
    """Centralized icon mapping system for the scraper"""
    
    # Food Type Icons with colors
    FOOD_ICONS = {
        # Meat Types
        'chicken': {'icon': 'mdi:food-drumstick', 'color': '#D4AF37'},
        'wings': {'icon': 'mdi:food-drumstick', 'color': '#D4AF37'},
        'breast': {'icon': 'mdi:food-drumstick', 'color': '#D4AF37'},
        'kjúklingur': {'icon': 'mdi:food-drumstick', 'color': '#D4AF37'},
        'kjúkling': {'icon': 'mdi:food-drumstick', 'color': '#D4AF37'},
        
        'beef': {'icon': 'mdi:food-steak', 'color': '#8B4513'},
        'nautakjöt': {'icon': 'mdi:food-steak', 'color': '#8B4513'},
        'steak': {'icon': 'mdi:food-steak', 'color': '#8B4513'},
        
        # Fast Food
        'burger': {'icon': 'fluent-emoji:hamburger', 'color': '#8B4513'},
        'pizza': {'icon': 'twemoji:pizza', 'color': '#FF6B35'},
        'sub': {'icon': 'mdi:food-sandwich', 'color': '#228B22'},
        'bátur': {'icon': 'mdi:food-sandwich', 'color': '#228B22'},
        'sandwich': {'icon': 'mdi:food-hot-dog', 'color': '#228B22'},
        'hotdog': {'icon': 'mdi:food-hot-dog', 'color': '#8B4513'},
        'wrap': {'icon': 'mdi:food-hot-dog', 'color': '#228B22'},
        'panini': {'icon': 'mdi:food-hot-dog', 'color': '#228B22'},
        
        # Sides
        'franskar': {'icon': 'mdi:food-french-fries', 'color': '#8B4513'},
        'pommes': {'icon': 'mdi:food-french-fries', 'color': '#8B4513'},
        'fries': {'icon': 'mdi:food-french-fries', 'color': '#8B4513'},
        'potato': {'icon': 'mdi:food-french-fries', 'color': '#8B4513'},
        'kartöflur': {'icon': 'mdi:food-french-fries', 'color': '#8B4513'},
        'rice': {'icon': 'mdi:rice', 'color': '#F4A460'},
        'hrísgrjón': {'icon': 'mdi:rice', 'color': '#F4A460'},
        'salad': {'icon': 'noto:green-salad', 'color': '#228B22'},
        'salat': {'icon': 'noto:green-salad', 'color': '#228B22'},
        'grænmeti': {'icon': 'mdi:food-apple', 'color': '#228B22'},
        'vegetables': {'icon': 'mdi:food-apple', 'color': '#228B22'},
        
        # Drinks
        'drykkur': {'icon': 'mdi:cup', 'color': '#4A90E2'},
        'gos': {'icon': 'mdi:cup', 'color': '#000000'},
        'soda': {'icon': 'mdi:bottle-soda-classic', 'color': '#4A90E2'},
        'kók': {'icon': 'mdi:bottle-soda-classic', 'color': '#8B0000'},
        'pepsi': {'icon': 'mdi:bottle-soda-classic', 'color': '#0066CC'},
        'cola': {'icon': 'mdi:bottle-soda-classic', 'color': '#8B0000'},
        'water': {'icon': 'mdi:cup-water', 'color': '#87CEEB'},
        'vatn': {'icon': 'mdi:cup-water', 'color': '#87CEEB'},
        'beer': {'icon': 'mdi:beer', 'color': '#FFD700'},
        'bjór': {'icon': 'mdi:beer', 'color': '#FFD700'},
        'wine': {'icon': 'mdi:glass-wine', 'color': '#8B0000'},
        'vín': {'icon': 'mdi:glass-wine', 'color': '#8B0000'},
        'coffee': {'icon': 'mdi:coffee', 'color': '#8B4513'},
        'kaffi': {'icon': 'mdi:coffee', 'color': '#8B4513'},
        'tea': {'icon': 'mdi:tea', 'color': '#228B22'},
        'te': {'icon': 'mdi:tea', 'color': '#228B22'},
        
        # Sauces
        'sauce': {'icon': 'game-icons:ketchup', 'color': '#FF4500'},
        'sósa': {'icon': 'game-icons:ketchup', 'color': '#FF4500'},
        'ketchup': {'icon': 'game-icons:ketchup', 'color': '#FF0000'},
        'mayo': {'icon': 'game-icons:ketchup', 'color': '#FFFFF0'},
        'mustard': {'icon': 'game-icons:ketchup', 'color': '#FFD700'},
        'sinnep': {'icon': 'game-icons:ketchup', 'color': '#FFD700'},
        
        # Desserts
        'ice-cream': {'icon': 'mdi:ice-cream', 'color': '#FF69B4'},
        'ís': {'icon': 'mdi:ice-cream', 'color': '#FF69B4'},
        'cake': {'icon': 'mdi:cake-variant', 'color': '#FF69B4'},
        'kaka': {'icon': 'mdi:cake-variant', 'color': '#FF69B4'},
        'cookie': {'icon': 'mdi:cookie', 'color': '#D2691E'},
        'kex': {'icon': 'mdi:cookie', 'color': '#D2691E'},
        'dessert': {'icon': 'mdi:ice-cream', 'color': '#FF69B4'},
        'eftirréttur': {'icon': 'mdi:ice-cream', 'color': '#FF69B4'},
        
        # Fish
        'fish': {'icon': 'mdi:fish', 'color': '#4A90E2'},
        'fiskur': {'icon': 'mdi:fish', 'color': '#4A90E2'},
        'salmon': {'icon': 'mdi:fish', 'color': '#FF6B6B'},
        'lax': {'icon': 'mdi:fish', 'color': '#FF6B6B'},
        'cod': {'icon': 'mdi:fish', 'color': '#4A90E2'},
        'þorskur': {'icon': 'mdi:fish', 'color': '#4A90E2'},
        'shrimp': {'icon': 'noto-v1:shrimp', 'color': '#FF6B6B'},
        'rækja': {'icon': 'noto-v1:shrimp', 'color': '#FF6B6B'},
        
        # Generic
        'main': {'icon': 'mdi:food', 'color': '#8B4513'},
        'side': {'icon': 'mdi:food-fork-drink', 'color': '#228B22'},
        'drink': {'icon': 'mdi:cup', 'color': '#4A90E2'},
        'snack': {'icon': 'mdi:food-variant', 'color': '#8B4513'},
        'fruit': {'icon': 'mdi:food-apple', 'color': '#228B22'},
        'ávöxtur': {'icon': 'mdi:food-apple', 'color': '#228B22'},
        'soup': {'icon': 'mdi:food-variant', 'color': '#8B4513'},
        'suppa': {'icon': 'mdi:food-variant', 'color': '#8B4513'},
        'curry': {'icon': 'mdi:food-variant', 'color': '#FF6B35'},
        'kebab': {'icon': 'mdi:food-variant', 'color': '#8B4513'},
        'quesadilla': {'icon': 'mdi:food-variant', 'color': '#FFD700'},
        'burrito': {'icon': 'mdi:food-variant', 'color': '#228B22'},
        'enchilada': {'icon': 'mdi:food-variant', 'color': '#FF6B35'},
        'fajita': {'icon': 'mdi:food-variant', 'color': '#FFD700'},
        'gyro': {'icon': 'mdi:food-variant', 'color': '#8B4513'},
        'shawarma': {'icon': 'mdi:food-variant', 'color': '#8B4513'},
        'falafel': {'icon': 'mdi:food-variant', 'color': '#228B22'},
        'dumpling': {'icon': 'mdi:food-variant', 'color': '#8B4513'},
        'springroll': {'icon': 'mdi:food-variant', 'color': '#228B22'},
        'eggroll': {'icon': 'mdi:food-variant', 'color': '#FFD700'},
        'tempura': {'icon': 'mdi:food-variant', 'color': '#FFD700'},
        'teriyaki': {'icon': 'mdi:food-variant', 'color': '#8B4513'},
        'stirfry': {'icon': 'mdi:food-variant', 'color': '#228B22'},
        'lo mein': {'icon': 'mdi:food-variant', 'color': '#8B4513'},
        'pad thai': {'icon': 'mdi:food-variant', 'color': '#FFD700'},
        'pho': {'icon': 'mdi:food-variant', 'color': '#8B4513'},
        'ramen': {'icon': 'mdi:food-variant', 'color': '#8B4513'},
        'udon': {'icon': 'mdi:food-variant', 'color': '#8B4513'},
        'soba': {'icon': 'mdi:food-variant', 'color': '#8B4513'},
        'bibimbap': {'icon': 'mdi:food-variant', 'color': '#FF6B35'},
        'bulgogi': {'icon': 'mdi:food-variant', 'color': '#8B4513'},
        'japchae': {'icon': 'mdi:food-variant', 'color': '#8B4513'},
        'kimchi': {'icon': 'mdi:food-variant', 'color': '#FF6B35'},
        'taco': {'icon': 'mdi:food-variant', 'color': '#FFD700'},
        'thai': {'icon': 'mdi:food-variant', 'color': '#FF6B35'},
        'noodlesoup': {'icon': 'mdi:food-variant', 'color': '#8B4513'},
        'noodles': {'icon': 'mdi:food-variant', 'color': '#8B4513'},
        'núðlur': {'icon': 'mdi:food-variant', 'color': '#8B4513'},
        'pasta': {'icon': 'mdi:food-variant', 'color': '#8B4513'},
        'sushi': {'icon': 'mdi:food-variant', 'color': '#FF6B6B'},
    }
    
    # Tag Icons (for offer tags)
    TAG_ICONS = {
        'Borgari': {'icon': 'fluent-emoji:hamburger', 'color': '#8B4513'},
        'Kjúklingur': {'icon': 'mdi:food-drumstick', 'color': '#D4AF37'},
        'Nautakjöt': {'icon': 'mdi:food-steak', 'color': '#8B4513'},
        'Pizza': {'icon': 'twemoji:pizza', 'color': '#FF6B35'},
        'Franskar': {'icon': 'mdi:food-french-fries', 'color': '#8B4513'},
        'Sub': {'icon': 'mdi:food-sandwich', 'color': '#228B22'},
        'Gos': {'icon': 'mdi:cup', 'color': '#000000'},
        'Drykkur': {'icon': 'mdi:cup', 'color': '#4A90E2'},
        'Fiskur': {'icon': 'mdi:fish', 'color': '#4A90E2'},
        'Rækjur': {'icon': 'noto-v1:shrimp', 'color': '#FF6B6B'},
        'Pasta': {'icon': 'mdi:food-variant', 'color': '#8B4513'},
        'Sushi': {'icon': 'mdi:food-variant', 'color': '#FF6B6B'},
        'Thai': {'icon': 'mdi:food-variant', 'color': '#FF6B35'},
        'Núðlur': {'icon': 'mdi:food-variant', 'color': '#8B4513'},
        'Núðlusúp': {'icon': 'mdi:food-variant', 'color': '#8B4513'},
        'Vegan': {'icon': 'mdi:food-variant', 'color': '#228B22'},
    }
    
    @classmethod
    def _load_food_icons(cls):
        """Load icon mappings from JSON and cache them"""
        if not hasattr(cls, '_icons_cache'):
            json_path = Path(__file__).resolve().parent.parent / 'food_icon_mapping.json'
            try:
                with open(json_path, 'r', encoding='utf-8') as f:
                    cls._icons_cache = json.load(f)
            except FileNotFoundError:
                cls._icons_cache = {}
        return cls._icons_cache
    
    @classmethod
    def get_food_icon(cls, food_type: str, category: str = None) -> Dict[str, str]:
        """Get icon and color for a food item using JSON mapping"""
        icons = cls._load_food_icons()
        normalized_type = food_type.lower().strip()
        normalized_category = category.lower().strip() if category else None

        # Try exact match first
        if normalized_type in icons:
            return icons[normalized_type]

        # Try category match
        if normalized_category and normalized_category in icons:
            return icons[normalized_category]

        # Try partial matches
        for key, mapping in icons.items():
            if normalized_type in key or key in normalized_type:
                return mapping

        # Default fallback
        return {'icon': 'mdi:food', 'color': '#8B4513'}
    
    @classmethod
    def generate_standardized_tags(cls, offer: Dict[str, Any]) -> List[Dict[str, str]]:
        """Generate standardized tags for an offer"""
        tags = []
        all_items = offer.get('food_items', [])
        
        # Chicken
        if any(item.get('type') == 'chicken' or 'kjúkling' in item.get('name', '').lower() 
               for item in all_items):
            tag_icon = cls.TAG_ICONS['Kjúklingur']
            tags.append({
                'label': 'Kjúklingur',
                'icon': tag_icon['icon'],
                'color': tag_icon['color']
            })
        
        # Beef
        if any(item.get('type') == 'beef' or 'nautakjöt' in item.get('name', '').lower() 
               for item in all_items):
            tag_icon = cls.TAG_ICONS['Nautakjöt']
            tags.append({
                'label': 'Nautakjöt',
                'icon': tag_icon['icon'],
                'color': tag_icon['color']
            })
        
        # Pizza
        if any(item.get('type') == 'pizza' or 'pizza' in item.get('name', '').lower() 
               for item in all_items):
            tag_icon = cls.TAG_ICONS['Pizza']
            tags.append({
                'label': 'Pizza',
                'icon': tag_icon['icon'],
                'color': tag_icon['color']
            })
        
        # French fries
        if any('franskar' in item.get('name', '').lower() or 'pommes' in item.get('name', '').lower() 
               for item in all_items):
            tag_icon = cls.TAG_ICONS['Franskar']
            tags.append({
                'label': 'Franskar',
                'icon': tag_icon['icon'],
                'color': tag_icon['color']
            })
        
        # Sub/Bátur
        if any('bátur' in item.get('name', '').lower() or 'sub' in item.get('name', '').lower() 
               for item in all_items):
            tag_icon = cls.TAG_ICONS['Sub']
            tags.append({
                'label': 'Sub',
                'icon': tag_icon['icon'],
                'color': tag_icon['color']
            })
        
        # Drink included - check if it's soda/gos
        drink_items = offer.get('drink_items', [])
        if drink_items:
            has_soda = any(
                'gos' in drink.get('name', '').lower() or 
                'soda' in drink.get('name', '').lower() or 
                'kók' in drink.get('name', '').lower() or
                'pepsi' in drink.get('name', '').lower()
                for drink in drink_items
            )
            if has_soda:
                tag_icon = cls.TAG_ICONS['Gos']
                tags.append({
                    'label': 'Gos',
                    'icon': tag_icon['icon'],
                    'color': tag_icon['color']
                })
            else:
                tag_icon = cls.TAG_ICONS['Drykkur']
                tags.append({
                    'label': 'Drykkur',
                    'icon': tag_icon['icon'],
                    'color': tag_icon['color']
                })
        
        return tags[:3]  # Limit to 3 tags
    
    @classmethod
    def enhance_food_item(cls, food_item: Dict[str, Any]) -> Dict[str, Any]:
        """Enhance a food item with proper icon and color"""
        food_type = food_item.get('type', '')
        category = food_item.get('category', '')
        name = food_item.get('name', '')
        
        # Get icon mapping
        icon_mapping = cls.get_food_icon(food_type, category)
        
        # Update the food item
        enhanced_item = food_item.copy()
        enhanced_item['icon'] = icon_mapping['icon']
        enhanced_item['icon_color'] = icon_mapping['color']
        
        return enhanced_item
    
    @classmethod
    def enhance_offer(cls, offer: Dict[str, Any]) -> Dict[str, Any]:
        """Enhance an offer with all necessary display information"""
        enhanced_offer = offer.copy()
        
        # Enhance all food items
        if 'food_items' in enhanced_offer:
            enhanced_offer['food_items'] = [
                cls.enhance_food_item(item) for item in enhanced_offer['food_items']
            ]
        
        if 'main_items' in enhanced_offer:
            enhanced_offer['main_items'] = [
                cls.enhance_food_item(item) for item in enhanced_offer['main_items']
            ]
        
        if 'side_items' in enhanced_offer:
            enhanced_offer['side_items'] = [
                cls.enhance_food_item(item) for item in enhanced_offer['side_items']
            ]
        
        if 'drink_items' in enhanced_offer:
            enhanced_offer['drink_items'] = [
                cls.enhance_food_item(item) for item in enhanced_offer['drink_items']
            ]
        
        if 'dessert_items' in enhanced_offer:
            enhanced_offer['dessert_items'] = [
                cls.enhance_food_item(item) for item in enhanced_offer['dessert_items']
            ]
        
        # Generate standardized tags
        enhanced_offer['standardized_tags'] = cls.generate_standardized_tags(enhanced_offer)
        
        # Add display properties
        enhanced_offer['display_properties'] = {
            'show_category_headers': False,  # Don't show aðalréttur, meðlæmi, etc.
            'show_fallback_info': False,    # Don't show "Upplýsingar" section
            'compact_view': True,           # Use compact food visualization
            'max_tags': 3                   # Maximum number of tags to show
        }
        
        return enhanced_offer 